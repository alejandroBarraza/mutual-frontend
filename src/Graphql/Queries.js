import { gql } from '@apollo/client';

//videos queries(always get last created)
export const GETVIDEOS = gql`
    query Getvideo {
        videos(limit: 1, sort: "created_at:desc") {
            created_at
            titulo
            descripcion
            links
        }
    }
`;

//statistic query(always get last 4 created)
export const GETSTATISTICS = gql`
    query GetStatistic {
        statistics(limit: 4, sort: "created_at:asc") {
            titulo
            descripcion
            created_at
        }
    }
`;

// get download content given limit and start =="offset" as arguments
export const GETDOWNLOADS_LIMIT_START = gql`
    query GetDownloads($start: Int!, $limit: Int!) {
        downloads(start: $start, limit: $limit) {
            id
            published_at
            titulo
            imagenes {
                name
                url
            }
        }
    }
`;

export const GETDOWNLOADS_SORTED_DESC = gql`
    query GetDownloads {
        downloads(sort: "created_at:desc") {
            id
            titulo
            descripcion
            archivo {
                url
            }
        }
    }
`;

export const GETNEWS_START_LIMIT = gql`
    query getNews($limit: Int!, $start: Int!) {
        news(limit: $limit, start: $start, sort: "created_at:desc") {
            id
            titulo
            descripcion
            tiempoLectura
            imagen {
                url
            }
        }
    }
`;

// get the footer elements
export const GET_ALL_FOOTER = gql`
    query getFooter {
        footer {
            selloSeguridad
            descripcion
            instagram
            facebook
            twitter
            youtube
        }
    }
`;
