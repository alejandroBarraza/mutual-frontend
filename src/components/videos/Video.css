.title-video {
  margin: 2rem 0 0.5rem 0;
  font-size: var(--font-size-h4);
  background: -webkit-linear-gradient(#607e06, #91b722);
  text-align: center;
  background-clip: text;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  /* color: var(--mutual-color); */
}
.description-video {
  font-size: var(--font-size-paragraph);
  padding: 1rem 0 1rem 0rem;
  color: var(--paragraph-color);
  line-height: 130%;
}

/* small  */
@media (min-width: 576px) {
  .title-video {
    margin: 3rem 0 0 0;
  }
  .description-video {
    padding: 2rem 0 2rem 0rem;
  }
}
/* medium  */

@media (min-width: 768px) {
  .title-video {
    margin: 3rem 0 0 0;
    text-align: center;
  }
  .description-video {
    padding: 2rem 0 2rem 0rem;
    text-align: center;
  }
}
/* large  */
@media (min-width: 992px) {
  .title-video {
    text-align: center;
    margin: 4rem 0 0 0;
    font-size: var(--font-size-h2);
  }
  .description-video {
    text-align: center;
    padding: 3rem 0 3rem 0rem;
  }
}
/* x-large  */
@media (min-width: 1200px) {
  .title-video {
    margin: 4rem 0 0 0;
    text-align: center;
    font-size: var(--font-size-h1);
  }
  .description-video {
    text-align: center;
    padding: 3rem 0 3rem 0rem;
  }
}
