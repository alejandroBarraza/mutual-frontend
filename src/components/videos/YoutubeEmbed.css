.container-video {
    overflow: hidden;
    padding-bottom: 56.25%;
    position: relative;
    height: 0;
    z-index: 0;
}
.container-video iframe {
    align-items: center;
    border-radius: 10px;
    height: 90%;
    justify-content: center;
    left: 0;
    margin: auto;
    position: absolute;
    top: 0;
    width: 100%;
}
