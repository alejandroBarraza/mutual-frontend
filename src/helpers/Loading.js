import React from 'react';
import { CircularProgress } from '@material-ui/core';
import '../container.css';

export const Loading = () => {
    return (
        <div className='container' style={{ textAlign: 'center' }}>
            <CircularProgress />
        </div>
    );
};
