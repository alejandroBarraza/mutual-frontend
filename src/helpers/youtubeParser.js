// get url yotube video and return id video
// source = https://stackoverflow.com/questions/3452546/how-do-i-get-the-youtube-video-id-from-a-url
export const youtube_parser = (url) => {
    const regExp = /^.*((youtu.be\/)|(v\/)|(\/u\/\w\/)|(embed\/)|(watch\?))\??v?=?([^#&?]*).*/;
    const match = url.match(regExp);
    return match && match[7].length === 11 ? match[7] : false;
};
