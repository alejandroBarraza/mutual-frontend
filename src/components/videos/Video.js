import React from 'react';
import { useQuery } from '@apollo/client';
import { YoutubeEmbed } from './YoutubeEmbed';
import { GETVIDEOS } from '../../Graphql/Queries';
import './Video.css';
import '../../container.css';
import { Loading } from '../../helpers/Loading';

export const Video = () => {
    const { data, loading, error } = useQuery(GETVIDEOS);
    if (loading) return <Loading />;
    if (error) return <p>{JSON.stringify(error)}</p>;

    return (
        <div className='container'>
            <div>
                <h2 className='title-video'>{data.videos[0].titulo}</h2>
                <p className='description-video'>{data.videos[0].descripcion}</p>
                <YoutubeEmbed embedId={data.videos[0].links} />
            </div>
        </div>
    );
};
