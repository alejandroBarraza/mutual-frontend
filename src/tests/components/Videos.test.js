afterEach(cleanup);
import { render, cleanup, waitFor } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { GETVIDEOS } from '../../Graphql/Queries';
import { Video } from '../../components/videos/Video';
import { youtube_parser } from '../../helpers/youtubeParser';
import '@wojtekmaj/enzyme-adapter-react-17';

const mocks = [
    {
        request: {
            query: GETVIDEOS,
            variables: { id: null },
        },
        result: {
            data: {
                videos: [
                    {
                        created_at: '2021-10-13T17:36:55.714Z',
                        descripcion: 'mutual',
                        links: 'https://www.youtube.com/watch?v=NrO0CJCbYLA',
                        titulo: 'mutual lo mejor de chile',
                    },
                ],
            },
        },
    },
];

const mocksError = [
    {
        request: {
            query: GETVIDEOS,
            variables: { id: null },
        },
        error: new Error('An error occurred'),
    },
];

describe('Testing <Videos/> component at error,loading and data fetched states.', () => {
    it('should render <Loading/> before graphql promise get resolved.', () => {
        const { container } = render(
            <MockedProvider mocks={mocks} addTypename={false}>
                <Video />
            </MockedProvider>
        );

        //select loading component root class.
        const loading = container.querySelector('.MuiCircularProgress-root');
        expect(loading).toBeInTheDocument();
    });

    it('should render the Error ui component in Error Fetch Data', async () => {
        const { container } = render(
            <MockedProvider mocks={mocksError} addTypename={false}>
                <Video />
            </MockedProvider>
        );
        await waitFor(() => new Promise((res) => setTimeout(res, 0)));
        const error = container.querySelector('.container-error');
        console.log(error.innerHTML);
        expect(error).toMatchSnapshot();
    });

    it('should match <Video/> component with Snaphshot', async () => {
        const { container } = render(
            <MockedProvider mocks={mocks} addTypename={false}>
                <Video />
            </MockedProvider>
        );
        await waitFor(() => new Promise((res) => setTimeout(res, 0)));
        expect(container).toMatchSnapshot();
    });

    it('should check if function youtoube Parse work properly given a youtube url.', () => {
        const url = youtube_parser('https://www.youtube.com/watch?v=aXzEmJs6_Qk');
        console.log(url);
        expect(url).toBe('aXzEmJs6_Qk');
    });
});
