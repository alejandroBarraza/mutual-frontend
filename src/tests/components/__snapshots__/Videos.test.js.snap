// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Videos/> component at error,loading and data fetched states. should match <Video/> component with Snaphshot 1`] = `
<div>
  <div
    class="container"
  >
    <h2
      class="title-video"
    >
      mutual lo mejor de chile
    </h2>
    <p
      class="description-video"
    >
      mutual
    </p>
    <div
      class="container-video"
    >
      <iframe
        allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
        allowfullscreen=""
        frameborder="0"
        height="480"
        src="https://www.youtube.com/embed/NrO0CJCbYLA"
        title="Embedded youtube"
        width="853"
      />
    </div>
  </div>
</div>
`;

exports[`Testing <Videos/> component at error,loading and data fetched states. should render the Error ui component in Error Fetch Data 1`] = `
<div
  class="container-error"
>
  <p>
    "An error occurred"
  </p>
</div>
`;
